beacon-chain/blockchain/service.go

```diff
@@ -94,6 +94,32 @@ var ErrMissingClockSetter = errors.New("blockchain Service initialized without a
 type blobNotifierMap struct {
 	sync.RWMutex
 	notifiers map[[32]byte]chan uint64
+	seenIndex map[[32]byte]map[uint64]struct{}
+}
+
+func (bn *blobNotifierMap) notifyIndex(root [32]byte, idx uint64) {
+	bn.Lock()
+	defer bn.Unlock()
+
+	// Initialize the map for the given root if it doesn't exist
+	if _, ok := bn.seenIndex[root]; !ok {
+		bn.seenIndex[root] = make(map[uint64]struct{})
```

Comment:
What if instead of a nested map you used `[fieldparams.MaxBlobsPerBlock]bool`? That forces you to check that the index isn't out of bounds / matches the size of the buffered channel.

Reply:
Also safer if you check that `idx < fieldparams.MaxBlobsPerBlock` to make sure you don't exceed channel's buffer and deadlock.